bug_view_inc.php

```diff
@@ -588,6 +590,39 @@
 	echo '</td></tr>';
 }
 
+# Attachments
+if( !empty( $t_result['issue']['attachments'] ) ) {
+	echo '<tr class="noprint">';
+	echo '<th class="bug-attach-tags category">', lang_get( 'files' ), '</th>';
+	echo '<td class="bug-attach-tags" colspan="5">';
+
+	$t_bug_activity_get_all_result = bug_activity_get_all( $f_issue_id, /* include_attachments */ true );
+	$t_activities = $t_bug_activity_get_all_result['activities'];
+	$t_security_token_attachments_delete = form_security_token( 'bug_file_delete' );
+
+	foreach( $t_activities as $t_activity ) {
+		if( $t_activity['type'] !== ENTRY_TYPE_ATTACHMENT ) {
+			continue;
+		}
+
+		$t_attachments = array();
+		if( isset( $t_activity['attachment'] ) ) {
+			$t_attachments[(int)($t_activity['attachment']['id'])] = $t_activity['attachment'];
+		}
+
+		foreach( $t_activity['attachments'] as $t_attachment ) {
+			$t_attachments[(int)$t_attachment['id']] = $t_attachment;
+		}
+
+		ksort( $t_attachments, SORT_NUMERIC );
+		foreach( $t_attachments as $t_attachment ) {
+			print_bug_attachment( $t_attachment, $t_security_token_attachments_delete );
+		}
+	}
```

Comment:
This yields
> SYSTEM WARNING: 'Undefined array key "attachments"' in '/home/dregad/dev/mantisbt/bug_view_inc.php' line 613
> SYSTEM WARNING: 'foreach() argument must be of type array|object, null given' in '/home/dregad/dev/mantisbt/bug_view_inc.php' line 613

Unless I'm mistaken, only activities of type _note_ have an `attachments` key, and these are ignored by the loop (line 604), so you could get rid of the offending inner foreach entirely.

The ksort and subsequent foreach do not make any sense, because each _attachment_ type activity can only have a single attachment.

So the loop boils down to this:
```php
	foreach( $t_activities as $t_activity ) {
		if( $t_activity['type'] === ENTRY_TYPE_ATTACHMENT ) {
			print_bug_attachment( $t_activity['attachment'], $t_security_token_attachments_delete );
		}
	}
```